src/ssmDocument/ssm/ssmServer.ts

```diff
@@ -0,0 +1,537 @@
+        return URL.resolve(resource, relativePath)
+    },
+}
+
+// create the JSON language service
+let languageService = getLanguageService({
+    workspaceContext,
+    contributions: [],
+    clientCapabilities: JsonLS.ClientCapabilities.LATEST,
+})
+
+// Create a text document manager.
+const documents = new TextDocuments(JsonLS.TextDocument)
+
+// Make the text document manager listen on the connection
+// for open, change and close text document events
+documents.listen(connection)
+
+let clientSnippetSupport = false
+let dynamicFormatterRegistration = false
+let hierarchicalDocumentSymbolSupport = false
+
+let foldingRangeLimitDefault = Number.MAX_VALUE
+let foldingRangeLimit = Number.MAX_VALUE
+let resultLimit = Number.MAX_VALUE
+
+// After the server has started the client sends an initialize request. The server receives
+// in the passed params the rootPath of the workspace plus the client capabilities.
+connection.onInitialize(
+    (params: InitializeParams): InitializeResult => {
+        languageService = getLanguageService({
+            workspaceContext,
+            contributions: [],
+            clientCapabilities: params.capabilities,
+        })
+
+        // tslint:disable: no-unsafe-any
+        function getClientCapability<T>(name: string, def: T) {
+            const keys = name.split('.')
+            let c: any = params.capabilities
+            for (let i = 0; c && i < keys.length; i++) {
+                if (!c.hasOwnProperty(keys[i])) {
+                    return def
+                }
+                c = c[keys[i]]
+            }
+
+            return c
+        }
+
+        clientSnippetSupport = getClientCapability('textDocument.completion.completionItem.snippetSupport', false)
+        dynamicFormatterRegistration =
+            getClientCapability('textDocument.rangeFormatting.dynamicRegistration', false) &&
+            typeof params.initializationOptions.provideFormatter !== 'boolean'
+        foldingRangeLimitDefault = getClientCapability('textDocument.foldingRange.rangeLimit', Number.MAX_VALUE)
+        hierarchicalDocumentSymbolSupport = getClientCapability(
+            'textDocument.documentSymbol.hierarchicalDocumentSymbolSupport',
+            false
+        )
+
+        // Need all letters to be trigger characters for YAML completion
+        const triggerCharacters = [
+            '"',
+            '-',
+            '$',
+            'a',
+            'b',
+            'c',
+            'd',
+            'e',
+            'f',
+            'g',
+            'h',
+            'i',
+            'j',
+            'k',
+            'l',
+            'm',
+            'n',
+            'o',
+            'p',
+            'q',
+            'r',
+            's',
+            't',
+            'u',
+            'v',
+            'w',
+            'x',
+            'y',
+            'z',
+        ]
+        // tslint:enable: no-unsafe-any
```

Comment:
Why enable here?

Reply:
I will move this up, so only `getClientCapability ` is inside the disabled range.